mlir/lib/Bindings/Python/DialectLinalg.cpp

```diff
@@ -20,6 +53,33 @@ static void populateDialectLinalgSubmodule(nb::module_ m) {
       nb::arg("op"),
       "Fill the region for `op`, which is assumed to be a builtin named Linalg "
       "op.");
+
+  m.def("isa_contraction_op", &mlirLinalgIsContractionOp,
+        "Checks if the given operation is a Linalg contraction operation.",
+        nb::arg("op"));
+
+  nb::class_<PyContractionDimensions>(m, "ContractionDimensions")
+      .def_prop_ro("batch",
+                   [](const PyContractionDimensions &self) {
+                     return convertDenseI32AttrToList(self.value.batch);
+                   })
+      .def_prop_ro("m",
+                   [](const PyContractionDimensions &self) {
+                     return convertDenseI32AttrToList(self.value.m);
+                   })
+      .def_prop_ro("n",
+                   [](const PyContractionDimensions &self) {
+                     return convertDenseI32AttrToList(self.value.n);
+                   })
+      .def_prop_ro("k", [](const PyContractionDimensions &self) {
+        return convertDenseI32AttrToList(self.value.k);
```

Comment:
```suggestion
      .def_prop_ro("batch",
                   [](const PyContractionDimensions &self) {
                     return self.value.batch;
                   })
      .def_prop_ro("m",
                   [](const PyContractionDimensions &self) {
                     return self.value.m;
                   })
      .def_prop_ro("n",
                   [](const PyContractionDimensions &self) {
                     return self.value.n;
                   })
      .def_prop_ro("k", [](const PyContractionDimensions &self) {
        return self.value.k;
```

see [above comment](https://github.com/llvm/llvm-project/pull/134935/files#r2034186175).

Note that when you remove the unnecessary `PyContractionDimensions` wrapper you'll be able to do 

```c++
      .def_prop_ro("batch", &MlirLinalgContractionDimensions::batch)
      .def_prop_ro("m", &MlirLinalgContractionDimensions::m)
      .def_prop_ro("n", &MlirLinalgContractionDimensions::n)
      .def_prop_ro("k", &MlirLinalgContractionDimensions::k);
```

Reply:
I tried to do this
```c
.def_prop_ro("m", &MlirLinalgContractionDimensions::m)
```

but I got a compilation error:
```
error: no matching function for call to ‘cpp_function<MlirLinalgContractionDimensions>(MlirAttribute MlirLinalgContractionDimensions::*&, ...)’
```

so here is the modified version I am trying to use as shown below:
```python
  nb::class_<MlirLinalgContractionDimensions>(m, "ContractionDimensions")
      .def_prop_ro("batch",
                   [](const MlirLinalgContractionDimensions &self) {
                     return self.batch;
                   })
      .def_prop_ro(
          "m",
          [](const MlirLinalgContractionDimensions &self) { return self.m; })
      .def_prop_ro(
          "n",
          [](const MlirLinalgContractionDimensions &self) { return self.n; })
      .def_prop_ro("k", [](const MlirLinalgContractionDimensions &self) {
        return self.k;
      });
```
Please let me know if I missed something here.

Reply:
ah it's because `prop_ro` is expecting `const` (`ro`). you could probably const cast the field using a really ugly function pointer type but this is fine too.